### PR TITLE
Support for datetime2 data type

### PR DESCRIPTION
**Pull request by tsegismont**
See #608

### CODE REVIEW

**Review by vietj — COMMENTED** [review]
I can see many reformatting changes because your IDE is configured differently for rewritten (type == String.class) as `(type==String.class)` can you remove those un-necessary changes ?

**Comment by tsegismont** [comment]
My IDE settings are correct, those lines were re-formatted because IDEA formats changed method bodies. It's just a few of them in a couple of methods.

**Comment by tsegismont** [comment]
@vietj anything else beyond formatting?

**Comment by vietj** [comment]
@tsegismont you seem to have peculiar IDE settings. Most vertx code base uses ` == ` , so that is a concern because the code you will be committing will always conflict and rewrite the code base, other committers (like me) will have their IDE reformat the other and it will be very confusing, so you should change your IDE settings to match this rule.

**Comment by tsegismont** [comment]
@vietj I use IDEA with default settings + the plugin for `.editorconfig` file. Not sure why but spaces around `==` got unchecked. I've updated my config and the PR.

Anything else beyond formatting?

**Review by vietj — APPROVED** [review]

**Comment by vietj** [comment]
can you rebase @tsegismont ?

**Comment by tsegismont** [comment]
@vietj just rebased on `master`. Can I merge when the GH checks are all green or do you want to take another look?

File: vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/ExtendedQueryCommandCodec.java [hunk 1]
```diff
@@ -207,6 +208,8 @@ private void encodeParamValue(ByteBuf payload, Object value) {
       encodeDateNParameter(payload, (LocalDate) value);
     } else if (value instanceof LocalTime) {
       encodeTimeNParameter(payload, (LocalTime) value, (byte) 6);
+    } else if (value instanceof LocalDateTime) {
+      encodeDateTimeNParameter(payload, (LocalDateTime) value, (byte) 6);
```
Comment (BillyYccc) [inline vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/ExtendedQueryCommandCodec.java:212]: Any opinions about the scale? I'm not sure if I ever got it right to hardcode it as 6 to not truncating any nano seconds in the `LocalTime` parameter.

Comment (tsegismont) [reply vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/ExtendedQueryCommandCodec.java:212]: I filed #752 to make sure we analyze and take any action needed.

**Pull request merged**